### PR TITLE
Add Bots to Room

### PR DESCRIPTION
**Pull request by tausifmuzaffar**
W-4329791

We want rooms to inherit the default bots for a roomType so we can identify the bots in a room.

### CODE REVIEW

File: db/model/room.js [hunk 1]
```diff
@@ -49,6 +49,11 @@ module.exports = function room(seq, dataTypes) {
       defaultValue: false,
       comment: 'Determines if room is still active',
     },
+    bots: {
+      type: dataTypes.ARRAY(dataTypes.STRING),
+      allowNull: true,
+      comment: 'Bots to be used in rooms',
```
Comment (kamilsmuga) [inline db/model/room.js]: "Bot names" instead of just "Bots"?

File: db/model/room.js [hunk 2]
```diff
@@ -75,22 +80,33 @@ module.exports = function room(seq, dataTypes) {
       },
     },
     hooks: {
-      afterCreate: (instance) => {
+
+      /**
+       * Ensures room gets default values from roomType, and puslish
```
Comment (kamilsmuga) [inline db/model/room.js]: "publishes" instead of "puslish"

File: tests/db/model/room/create.js [hunk 3]
```diff
@@ -67,6 +70,28 @@ describe('tests/db/model/room/create.js >', () => {
...
+      const rt = v.getStandard();
+      rt.bots = [bots.name];
+      return RoomType.create(rt);
+    })
+    .then((roomType) => {
+      const room = u.getStandard();
+      room.type = roomType.id;
+      return Room.create(room);
+    })
+    .then((o) => {
```
Comment (kamilsmuga) [inline tests/db/model/room/create.js]: "room" instead of "o" for better clarity?

File: db/model/room.js [hunk 4]
```diff
@@ -75,22 +80,33 @@ module.exports = function room(seq, dataTypes) {
...
         const RoomType = seq.models.RoomType;
         return RoomType.findById(instance.type)
         .then((roomType) => {
           instance.settings = roomType.settings;
+          instance.bots = roomType.bots;
           const changedKeys = Object.keys(instance._changed);
           const ignoreAttributes = ['isDeleted'];
-          return realTime.publishObject(instance.toJSON(), roomEventNames.add, changedKeys,
-              ignoreAttributes);
+          return realTime.publishObject(instance.toJSON(), roomEventNames.add,
```
Comment (iamigo) [inline db/model/room.js]: Seems weird to publish object BEFORE create. What if there's an error in the creation itself? Why are you doing it this way?

**Comment by tausifmuzaffar** [comment]
@pallavi2209 is this ready to merge?

**Comment by tausifmuzaffar** [comment]
@pallavi2209  Ithink i added the wrong Pallavi to the review before?

**Comment by pallavi2209** [comment]
@tausifmuzaffar Please go ahead and merge this once all the tests pass.

**Pull request merged**